tests/Model/CodecCursorFunctionalTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace MongoDB\Tests\Model;
+
+use MongoDB\BSON\Document;
+use MongoDB\Codec\DocumentCodec;
+use MongoDB\Model\CodecCursor;
+use MongoDB\Tests\FunctionalTestCase;
+
+class CodecCursorFunctionalTest extends FunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->dropCollection($this->getDatabaseName(), $this->getCollectionName());
+    }
+
+    public function testSetTypeMap(): void
+    {
+        $collection = self::createTestClient()->selectCollection($this->getDatabaseName(), $this->getCollectionName());
+        $cursor = $collection->find();
+
+        $codecCursor = CodecCursor::fromCursor($cursor, $this->createCodec());
+
+        $this->expectWarning();
+        $this->expectWarningMessage('Discarding type map for MongoDB\Model\CodecCursor::setTypeMap');
+
+        $codecCursor->setTypeMap(['root' => 'array']);
+    }
+
+    private function createCodec(): DocumentCodec
+    {
+        return new class implements DocumentCodec {
```

Comment:
This doesn't look like a meaningful implementation. Since `testSetTypeMap()` only calls `setTypeMap()`, perhaps you can just pass a mock DocumentCodec to `CodecCursor::fromCursor()`? I noted that the `$codec` property is only used in `current()`, which won't be called by the test.

Reply:
I keep forgetting about mocks, but you're correct. Changed accordingly.